modules/login/jsx/passwordExpiry.js

```diff
@@ -0,0 +1,185 @@
+    });
+    fetch(
+      url, {
+        method: 'POST',
+        headers: {
+          'Accept': 'application/json, text/plain, */*',
+          'Content-Type': 'application/x-www-form-urlencoded',
+        },
+        body: send,
+      }
+    ).then((response) => response.json())
+      .then(
+        (data) => {
+          if (data.error) {
+            // error - message.
+            const state = Object.assign({}, this.state);
+            state.form.error.toggle = true;
+            state.form.error.message = data.error;
+            this.setState(state);
+          } else {
+            // success - refresh page and user is logged in.
+            window.location.href = window.location.origin;
+          }
+        }).catch((error) => {
+      // no error should happen.
+    });
+  }
+
+  /**
+   * @return {DOMRect}
+   */
+  render() {
+    const error = this.state.form.error.toggle ? (
+      <StaticElement
+        text={this.state.form.error.message}
+        class={'col-xs-12 col-sm-12 col-md-12 text-danger'}
+      />
+    ) : null;
+    const success = !this.state.request ? (
+      <div>
+        <div>
+          <p><b>Password Strength Rules</b></p>
+          <ul>
+            <li>The password must be at least 8 characters long</li>
+            <li>The password must contain at least 1 letter, 1 number and 1 character from !@#$%^*()</li>
+            <li>The password and the user name must not be the same</li>
+            <li>The password and the email address must not be the same</li>
+          </ul>
+        </div>
+        <FormElement
+          name={'form1'}
+          action={''}
+          id={'form'}
+          fileUpload={'false'}
+          onSubmit={this.handleSubmit}
+        >
+          <TextboxElement
+            name={'password'}
+            value={this.state.form.value.password}
+            onUserInput={this.setForm}
+            class={'col-sm-12'}
+            required={true}
+            type={'password'}
+            placeholder={'New Password'}
+          />
+          <TextboxElement
+            name={'confirm'}
+            value={this.state.form.value.confirm}
+            onUserInput={this.setForm}
+            class={'col-sm-12'}
+            required={true}
+            type={'password'}
+            placeholder={'Confirm Password'}
+          />
+          {error}
+          <ButtonElement
+            label={'Save'}
+            type={'submit'}
+            columnSize={'col-sm-12'}
+            buttonClass={'btn btn-primary btn-block'}
+          />
+        </FormElement>
+        <a onClick={() => window.location.href = window.location.origin}
+           style={{cursor: 'pointer'}}>Back to login page</a>
+      </div>
+    ) : (
+      <div className={'success-message'}>
+        <h1>Thank you!</h1>
+        <p>Your request for an account has been received successfully.</p>
+        <a onClick={() => window.location.href = window.location.origin}
```

Comment:
@maltheism I think we should also add "Please contact your project administrator to activate this account" to avoid confusions to the user.

Reply:
@maltheism this still needs to be implemented?

Reply:
is my above comment still relevant?